src/test/java/uk/gov/ida/verifylocalmatchingserviceexample/dataaccess/TestDataSourceRule.java

```diff
@@ -0,0 +1,39 @@
+package uk.gov.ida.verifylocalmatchingserviceexample.dataaccess;
+
+import org.flywaydb.core.Flyway;
+import org.h2.jdbcx.JdbcConnectionPool;
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.core.Jdbi;
+import org.junit.rules.ExternalResource;
+
+public class TestDataSourceRule extends ExternalResource {
+    private JdbcConnectionPool connectionPool;
+    private Jdbi jdbi;
+
+    @Override
+    protected void before() throws Throwable {
+        connectionPool = JdbcConnectionPool.create("jdbc:h2:mem:test;MODE=PostgreSQL", "", "");
+        jdbi = Jdbi.create(connectionPool);
+        Handle handle = jdbi.open();
+
+        handle.begin();
+        setUpDatabase(connectionPool);
+        handle.commit();
+    }
+
```

Comment:
These tests are for checking if the SQL works as expected. I have used flyway migration for creating the table schema but the migration also populates data. As long as we don't depend on data created by migration I think we can use the migrations for these tests. Any thoughts on this ?

Reply:
Keerthana and I had a chat about this, and we think it's a good idea to run the migrations in the test set up.

In particular we think it's a good thing that if we change the schema in the migrations the tests will check that the old DAOs work with the new schema.

It's a little 😕 that the migrations insert a bunch of rows, but I think it's fine to pay that small price to ensure the schema is correct.